src/main.cc

```diff
@@ -1440,6 +1440,14 @@ StartUsingConfig()
 int
 SquidMain(int argc, char **argv)
 {
+    // We must register all modules before the first RunRegisteredHere() call.
+    // We do it ASAP/here so that we do not need to move this code when we add
+    // earlier hooks to the RegisteredRunner API. This collection of
+    // registration calls is not a RegisteredRunner "event" in itself.
+#if HAVE_AUTH_MODULE_NTLM
+    CallRunnerRegistrator(NtlmAuthRr);
+#endif
```

Comment:
Could we push it out to its own InitialiseSubmodules function?
Not a big deal but would make intent clearer

Reply:
Good idea! Done in commit 947b372.

> InitialiseSubmodules

I also emphasized that these module registrations do not _initialize_ those modules. We are (or should be), essentially, remembering a pointer to initialization (and, in general, event handling) code here, not calling the code behind that pointer.